windows-agent/internal/proservices/landscape/executor_test.go

```diff
@@ -167,16 +204,36 @@ func TestInstall(t *testing.T) {
 				name: testDistroAppx,
 			}
 
-			if tc.appxDoesNotExist {
+			if tc.appxDoesNotExist || tc.sendRootfsURL != "" {
 				// WSLMock Install only accepts ubuntu-22.04
 				settings.name = wsltestutils.RandomDistroName(t)
 			}
 
-			if tc.distroAlredyInstalled {
+			if tc.distroAlreadyInstalled {
 				settings.install = true
 			}
 
-			testReceiveCommand(t, settings,
+			// Here we depend on implementation details to increase test coverage :see_no_evil:
+			home := t.TempDir()
+			if tc.breakVhdxDir {
+				err := os.MkdirAll(filepath.Join(home, "WSL"), 0700)
+				require.NoError(t, err, "Setup: creating destination dir shouldn't fail")
+				f, err := os.Create(filepath.Join(home, "WSL", settings.name))
+				require.NoError(t, err, "Setup: breaking the destination dir shouldn't fail")
+				f.Close()
+			}
+
+			if tc.breakTarFile {
+				err := os.MkdirAll(filepath.Join(os.TempDir(), settings.name, settings.name+".tar.gz"), 0700)
+				require.NoError(t, err, "Setup: breaking the destination tarball shouldn't fail")
+			}
+
+			if tc.breakTempDir {
+				_, err := os.Create(filepath.Join(os.TempDir(), settings.name))
```

Comment:
`os.TempDir()` is probably not good in a test, but I'm unsure if we can get around unless we add support for injecting a custom temporary directory? I don't think this realistically is that big a problem since its 1. a temporary dir in the OS temp directory, and 2. its name should be unique to the test anyway since we use `settings.name`.

Reply:
As long as `settings.name` doesn't conflict we should be fine.

Reply:
Ah, see my previous comment :) I don’t see why we don’t use `t.Tempdir` TBH, as it will ensure we don’t touch the host system when running the tests and potentially leaving some leftover in case of breakage in the middle of a test?

Reply:
`installFromURL` would then need to have a way to use the test temp dir, correct?

Reply:
Yeah we pass `homeDir` to it and inject it in the test case with a temporary directory. `installFromURL` uses `filepath.Join(os.TempDir(), distro.Name())` as a temporary download path so we'd probably also need to add the ability to inject a custom temp dir if we are to use `t.TempDir()`.

Reply:
Right, and it will make more sense from a system perspective that we can control over where the temp download is processed.